eras/shelley/impl/src/Cardano/Ledger/Shelley/API/Validation.hs

```diff
@@ -94,6 +97,14 @@ class
     NewEpochState era ->
     Block (BHeaderView (EraCrypto era)) era ->
     m (EventReturnType ep (EraRule "BBODY" era) (NewEpochState era))
+  default applyBlockOpts ::
+    forall ep m.
+    (EventReturnTypeRep ep, MonadError (BlockTransitionError era) m, EraGovernance era) =>
```

Comment:
Instead of adding these `default` instances for `applyBlockOpts` and `reapplyBlock` you can add `EraGovernance` as the superclass to `ApplyBlock`, because it is a class that every era has to satisfy.

Reply:
GHC 9.6.2 was giving warnings about cyclical superclasses, so I figured I'll make it happy

Reply:
Yeah, that's weird. Anyways, nothing wrong with the default instances